### DISCUSSION TITLE
Bypass token validation

### DISCUSSION THREAD

**Discussion by crisclacerda** [PostgREST/postgrest#2161] [Q&A] [answered]
I am running postgrest behind an api gateway that already validates the token signature. 
I only want to check the role from the token without specifing the jwt secret. Since I am rotating my keys and postgrest do not support fetching the key from an api, I would be ok to simply bypass it...

**Comment by wolfgangwalther** [comment, marked as answer]
Can you do one of the following?

- Remove the `Authorization` header from the request in the gateway
- Add a custom header with the role extracted from the token to the request
- Run postgrest with a `db-pre-request` function which will then set the role from the header

or

- Replace the `Authorization` header with another header with the same token, but a different signature at the api gateway
- Use a simple, non-rotating secret for that internal signature, which you set `jwt-secret` to

**Reply by wolfgangwalther** [reply]
Also related:
- https://github.com/PostgREST/postgrest/issues/1130
- https://github.com/PostgREST/postgrest/issues/1402#issuecomment-557202233
- https://github.com/PostgREST/postgrest/issues/1727

**Comment by crisclacerda** [comment]
Great thanks for the help. I like the first solution and it works fine.

**Reply by wolfgangwalther** [reply]
Just make sure you also remove any of those custom headers that you are using internally from the original request, so that nobody can break in.
